### PR TITLE
Allow further service check property to be specified when registering a service

### PR DESCRIPTION
**Pull request by gboder**
Added additional Consul HTTP API parameter to detail the service check registration.

Reference to <https://www.consul.io/docs/discovery/services> and this <https://www.consul.io/docs/discovery/checks> page of Consul's documentation.

We are going to use those options to allow specify distinct host http header and host address values.

Sample definition
```csharp
new AgentServiceCheck{
	HTTP = $"http://srv1.servers.contoso.com/weatherforecast",
	Timeout = TimeSpan.FromMilliseconds(100),
	Interval = TimeSpan.FromSeconds(1),
	Header = new Dictionary<string, List<string>>{ {"Host",new List<string>{"app.contoso.com"} } },
	Method = "GET",
}
```

### CODE REVIEW

**Comment by gboder** [comment]
I'll give a look at the tests' result...

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,14 @@
 # Changelog
 
+## 1.6.1.3
```
Comment (marcin-krystianc) [inline CHANGELOG.md:3]: I'm not sure why did you decide to include two versions in the changelog ?
With your PR `v1.6.1.2` would never be released, so there is no point including it in the changelog.

File: Consul/Consul.csproj [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <VersionPrefix>1.6.1.1</VersionPrefix>
+    <VersionPrefix>1.6.1.3</VersionPrefix>
```
Comment (marcin-krystianc) [inline Consul/Consul.csproj:4]: I would suggest using version `1.7.0`:
- Three-part version number is more common versioning scheme for NuGet, https://docs.microsoft.com/en-us/nuget/concepts/package-versioning#version-basics
- According to the semantic versioning rules (https://semver.org/spec/v2.0.0.html) you should bump the `Minor` version:
```
...
- Bug fixes not affecting the API increment the patch version,
- Backwards compatible API additions/changes increment the minor version
- Backwards incompatible API changes increment the major version.
...
```

File: Consul/Agent.cs [hunk 3]
```diff
@@ -227,6 +218,16 @@ public class AgentCheckRegistration : AgentServiceCheck
...
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public string ID { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public string Name { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
+        public string Notes { get; set; }
+
```
Comment (marcin-krystianc) [inline Consul/Agent.cs:230]: 👍

**Review by marcin-krystianc — COMMENTED** [review]
Thanks @gboder for making this change.
Changes are good, just left a comment regarding the changelog and versioning.

File: CHANGELOG.md (hunk 1 above)
Comment (gboder) [reply CHANGELOG.md:3]: Right - dropping this entry and merging with the other changes. Also adjusting change log towards version '1.7.0'.

File: Consul/Consul.csproj (hunk 2 above)
Comment (gboder) [reply Consul/Consul.csproj]: I was also wondering a bit about the 4-digits notation, I only wanted to stay on the same pattern. Now if we think about three digits its for sure a `1.7.0`.

Let's follow the NuGet paradigm. I'll update to `1.7.0`

**Review by marcin-krystianc — APPROVED** [review]

**Comment by marcin-krystianc** [comment]
Hi @gboder ,

I reverted your version change back to the `v1.6.1.1`. I didn't realise that we have a rule to keep the version of `Consul.NET` to match the supported API version of `Consul' . Now the information about this is added to the [README](https://github.com/G-Research/consuldotnet/blob/master/README.md#versioning).

**Pull request merged**